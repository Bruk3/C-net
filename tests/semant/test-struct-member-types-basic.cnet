

struct pid {
	int id;
};

struct person {
	struct pid x;
	struct person partner;
};

int main() {
	struct person s;


	s; /* type struct person */
	s.x; /* type struct pid */
	s.x.id; /* type int */
	s.partner; /* type struct person */
	s.partner.partner; /* type struct person */
	s.partner.partner.x.id; /* type int */


	/* s.partner.partner.x.partner; ERROR */
}
