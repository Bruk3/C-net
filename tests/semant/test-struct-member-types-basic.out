struct pid {
	int id;
};

struct person {
	struct pid x;
	struct person partner;
};

int main()
{
struct person s;
(struct person : s);
(struct pid : s.x);
(int : s.x.id);
(struct person : s.partner);
(struct person : s.partner.partner);
(int : s.partner.partner.x.id);
}

